{
  float[] x={0.0f};
  boolean thrown=false;
  try {
    Array.setFloat(x,0,(float)1);
  }
 catch (  Exception e) {
    fail("Exception during get test : " + e.getMessage());
  }
  assertEquals("Get returned incorrect value",1,Array.getFloat(x,0),0.0);
  try {
    Array.setFloat(new Object(),0,(float)9);
  }
 catch (  IllegalArgumentException e) {
    thrown=true;
  }
  if (!thrown) {
    fail("Passing non-array failed to throw exception");
  }
  thrown=false;
  try {
    Array.setFloat(x,4,(float)9);
  }
 catch (  ArrayIndexOutOfBoundsException e) {
    thrown=true;
  }
  if (!thrown) {
    fail("Invalid index failed to throw exception");
  }
  thrown=false;
  try {
    Array.setFloat(null,0,0);
  }
 catch (  NullPointerException e) {
    thrown=true;
  }
  if (!thrown) {
    fail("Null argument failed to throw NPE");
  }
}
